astropy/nddata/bitmask.py

```diff
@@ -0,0 +1,411 @@
+"""
+A module that provides functions for manipulating bitmasks and data quality
+(DQ) arrays.
+
+"""
+
+import sys
+import warnings
+import six
+import numpy as np
+
+
+__all__ = ['bitfield_to_boolean_mask', 'interpret_bit_flags', 'is_bit_flag']
+
+
+INT_TYPE = (int, long,) if sys.version_info < (3,) else (int,)
+MAX_UINT_TYPE = np.maximum_sctype(np.uint)
+SUPPORTED_FLAGS = int(np.bitwise_not(
+    0, dtype=MAX_UINT_TYPE, casting='unsafe'
+))
+
+
+def is_bit_flag(n):
+    """
+    Verifies if the input number is a bit flag (i.e., an integer number that is
+    an integer power of 2).
+
+    Parameters
+    ----------
+    n : int
+        A positive integer number. Non-positive integers are considered not to
+        be "flags".
+
+    Returns
+    -------
+    bool
+        ``True`` if input ``n`` is a bit flag and ``False`` if it is not.
+
+    """
+    if n < 1:
+        return False
+
+    return bin(n).count('1') == 1
+
+
+def _is_int(n):
+    return (
+        (isinstance(n, INT_TYPE) and not isinstance(n, bool)) or
+        (isinstance(n, np.generic) and np.issubdtype(n, np.integer))
+    )
+
+
+def interpret_bit_flags(bit_flags, flip_bits=None):
+    """
+    Converts input bit flags to a single integer value (bitmask) or `None`.
+
+    When input is a list of flags (either a Python list of integer flags or a
+    sting of comma- or '+'-separated list of flags), the returned bitmask
+    is obtained by summing input flags.
+
+    .. note::
+        In order to flip the bits of the returned bitmask,
+        for input of `str` type, prepend '~' to the input string. '~' must
+        be prepended to the *entire string* and not to each bit flag! For
+        input that is already a bitmask or a Python list of bit flags, set
+        `flip_bits` for `True` in order to flip the bits of the returned
+        bitmask.
+
+    Parameters
+    ----------
+    bit_flags : int, str, list, None
+        An integer bitmask or flag, `None`, a string of comma- or
+        '+'-separated list of integer bit flags, or a Python list of integer
+        bit flags. If `bit_flags` is a `str` and if it is prepended with '~',
```

Comment:
`bit_flags` needs double back-ticks around it like this (sphinx interprets single backticks as links, not literal text like github/markdown does):
```
``bit_flags``
```